### PR TITLE
Updated component combobox

### PR DESCRIPTION
**Pull request by Hodgdonc**
Component combobox now includes the name of the component in addition to the ID.

### CODE REVIEW

**Review by Graham-Berry — COMMENTED** [review]
If I change the name of a component, then update, the prior name of the component stays in the combobox. Can this information be re-checked and updated after update?
image

File: loader.py [hunk 1]
```diff
@@ -69,6 +69,12 @@ def copyObjTemplate(self, _id):
     def getObjIDs(self):
         return list(self.obj_templates.keys())
```
Comment (Graham-Berry) [inline loader.py:70]: If I change the name of a component, then update, the prior name of the component stays in the combobox. Can this information be re-checked and updated after update?

Comment (Hodgdonc) [reply loader.py:70]: I believe this worked in an earlier commit, trying to figure out what broke it.

Comment (Hodgdonc) [reply loader.py:70]: The name is only updated in the combo box when you also update the ID value due to an if statement. Currently fixing it

Comment (Hodgdonc) [reply loader.py:70]: I believe my latest commit fixed the issue for both components and objects.

**Review by Graham-Berry — APPROVED** [review]
Yahurrrr

File: ui_advanced_config.py [hunk 2]
```diff
@@ -1161,20 +1168,21 @@ def update_teams_json(self):
...
-            and self.componentsIDEntry.get() != self.selectComponentCombo.get()
+            and self.componentsIDEntry.get()
+            != self.selectComponentCombo.get().split(":")[0]
         ):
             showwarning(
                 title="Warning",
                 message="The ID you are trying to use is already in use by another component. Please use another ID.",
             )
         else:
+            print("a")
```
Comment (sophiekummer) [inline ui_advanced_config.py]: remove this

File: ui_advanced_config.py [hunk 3]
```diff
@@ -1161,20 +1168,21 @@ def update_teams_json(self):
...
             if (
                 self.componentsIDEntry.get() != ""
                 and self.componentsNameEntry.get() != ""
                 and self.componentsTypeAttr1Entry.get() != ""
-                and self.componentsTypeAttr2Entry.get() != ""
-                and self.componentsTypeAttr3Entry != ""
             ):
+                print("b")
```
Comment (sophiekummer) [inline ui_advanced_config.py]: remove this

File: ui_advanced_config.py [hunk 4]
```diff
@@ -1286,19 +1294,56 @@ def update_components_json(self):
                         ),
                     )
 
+                if self.currentComponentData.getData("name") != self.componentData[
+                    self.selectComponentCombo.get().split(":")[0]
+                ].getData("name"):
+                    print("c")
```
Comment (sophiekummer) [inline ui_advanced_config.py]: remove this

File: ui_advanced_config.py [hunk 5]
```diff
@@ -1161,20 +1168,21 @@ def update_teams_json(self):
     def update_components_json(self):
```
Comment (sophiekummer) [inline ui_advanced_config.py:1165]: i can't update a components id anymore. if i try, the json and combo box do not update and there is the following error in the terminal:
Screen Shot 2024-04-02 at 1 52 23 PM

File: ui_advanced_config.py [hunk 6]
```diff
@@ -1161,20 +1168,21 @@ def update_teams_json(self):
...
+            print("a")
             if (
                 self.componentsIDEntry.get() != ""
                 and self.componentsNameEntry.get() != ""
                 and self.componentsTypeAttr1Entry.get() != ""
-                and self.componentsTypeAttr2Entry.get() != ""
-                and self.componentsTypeAttr3Entry != ""
             ):
+                print("b")
                 print(self.currentComponentData)
```
Comment (sophiekummer) [inline ui_advanced_config.py:1181]: remove this

File: ui_advanced_config.py [hunk 7]
```diff
@@ -1286,19 +1294,56 @@ def update_components_json(self):
...
+                        [
+                            self.currentCompIDs[comp_idx].split(":")[0],
+                            self.currentComponentData.getData("name"),
+                        ]
+                    )
+                    self.selectComponentCombo.configure(values=self.componentIDs)
+                    self.selectComponentCombo.current(len(self.componentIDs) - 1)
                 with open("settings/components.json", "r") as f:
                     componentJSON = json.load(f)
+                print("d")
```
Comment (sophiekummer) [inline ui_advanced_config.py]: remove this

File: ui_advanced_config.py [hunk 8]
```diff
@@ -1286,19 +1294,56 @@ def update_components_json(self):
...
                 if (
                     self.currentComponentData.getData("id")
-                    != self.selectComponentCombo.get()
+                    != self.selectComponentCombo.get().split(":")[0]
                 ):
-                    if self.selectComponentCombo.get() in componentJSON:
-                        componentJSON.pop(self.selectComponentCombo.get())
-                    if self.selectComponentCombo.get() in self.componentData:
-                        self.componentData.pop(self.selectComponentCombo.get())
+                    print("e")
```
Comment (sophiekummer) [inline ui_advanced_config.py]: remove this

File: ui_advanced_config.py [hunk 9]
```diff
@@ -1366,21 +1411,52 @@ def update_objects_json(self):
...
+                    self.objectIDs[obj_idx] = ": ".join(
+                        [
+                            self.objectIDs[obj_idx].split(":")[0],
+                            self.currentObjectData.getData("name"),
+                        ]
+                    )
+                    self.selectObjectsCombo.configure(values=self.objectIDs)
+                    self.selectObjectsCombo.current(len(self.objectIDs) - 1)
+
                 print(self.currentObjectData.getJSONView())
```
Comment (sophiekummer) [inline ui_advanced_config.py]: remove this

File: ui_advanced_config.py [hunk 10]
```diff
@@ -1366,21 +1411,52 @@ def update_objects_json(self):
...
                 with open("settings/objects.json", "r") as f:
                     objectJSON = json.load(f)
 
                 print(self.selectObjectsCombo.get())
```
Comment (sophiekummer) [inline ui_advanced_config.py:1402]: remove this

File: ui_advanced_config.py [hunk 11]
```diff
@@ -1429,12 +1505,27 @@ def update_maps_json(self):
                 with open("settings/maps.json", "r") as f:
                     mapJSON = json.load(f)
                     print(self.currentMapData.data)
-                    mapJSON[self.selectMapsCombo.get()] = self.currentMapData.data
+                if self.mapsIDEntry.get() != self.selectMapsCombo.get():
+                    print(self.selectMapsCombo.get())
```
Comment (sophiekummer) [inline ui_advanced_config.py:1490]: remove this

File: ui_advanced_config.py [hunk 12]
```diff
@@ -1429,12 +1505,27 @@ def update_maps_json(self):
...
+                        mapJSON.pop(self.selectMapsCombo.get())
+                    if self.selectMapsCombo.get() in self.mapData:
+                        self.mapData.pop(self.selectMapsCombo.get())
+                    self.mapIDs.pop(self.selectMapsCombo.current())
+                    self.mapIDs.append(self.mapsIDEntry.get())
+                    self.selectMapsCombo.configure(values=self.mapIDs)
+                    self.selectMapsCombo.current(len(self.mapIDs) - 1)
+
+                self.mapData[self.mapsIDEntry.get()] = self.currentMapData
+                print(mapJSON)
```
Comment (sophiekummer) [inline ui_advanced_config.py:1501]: remove this

File: ui_advanced_config.py [hunk 13]
```diff
@@ -1429,12 +1505,27 @@ def update_maps_json(self):
...
+                mapJSON[self.mapsIDEntry.get()] = self.currentMapData.data
                 f.close()
 
                 with open("settings/maps.json", "w") as f:
                     json.dump(mapJSON, f, indent=4)
                 f.close()
+                print(self.mapData)
```
Comment (sophiekummer) [inline ui_advanced_config.py:1508]: remove this

File: ui_advanced_config.py [hunk 14]
```diff
@@ -1429,12 +1505,27 @@ def update_maps_json(self):
...
+                print(self.currentMapData.data)
```
Comment (sophiekummer) [inline ui_advanced_config.py:1509]: remove this

File: ui_advanced_config.py [hunk 15]
```diff
@@ -545,7 +549,10 @@ def init_entry_widgets(self):
         # OBJECT
         self.objectData = self.ldr.obj_templates
         self.objectIDs = self.ldr.getObjIDs()
-        self.currentObjectData = self.objectData[self.objectIDs[0]]
+        self.objectNames = self.ldr.getObjNames()
+        for i in range(len(self.objectIDs)):
+            self.objectIDs[i] = self.objectIDs[i] + ": " + self.objectNames[i]
+        self.currentObjectData = self.objectData[self.objectIDs[0].split(":")[0]]
         print(self.currentObjectData.getSelfView())
```
Comment (sophiekummer) [inline ui_advanced_config.py:556]: remove this

File: ui_advanced_config.py [hunk 16]
```diff
@@ -783,7 +790,7 @@ def change_objects_entry_widgets(self, event=None):
         if self.answer:
             currentObject = self.selectObjectsCombo.get()
             print(currentObject)
```
Comment (sophiekummer) [inline ui_advanced_config.py:789]: remove this

File: ui_advanced_config.py [hunk 17]
```diff
@@ -1429,12 +1505,27 @@ def update_maps_json(self):
                 with open("settings/maps.json", "r") as f:
                     mapJSON = json.load(f)
                     print(self.currentMapData.data)
```
Comment (sophiekummer) [inline ui_advanced_config.py:1488]: remove this

File: ui_advanced_config.py [hunk 18]
```diff
@@ -1326,7 +1371,7 @@ def update_components_json(self):
     def update_objects_json(self):
```
Comment (sophiekummer) [inline ui_advanced_config.py:1355]: i can't update a objects id anymore. if i try, the json and combo box do not update and there is the following error in the terminal:
Screen Shot 2024-04-02 at 2 03 45 PM

File: ui_advanced_config.py [hunk 19]
```diff
@@ -1483,7 +1574,7 @@ def create_component(self):
                 "Component ID", "Please enter an ID for the new component."
```
Comment (sophiekummer) [inline ui_advanced_config.py:1555]: if i make a new component, the name is never added to the combobox, and it does not appear that the component is added to the json
Screen Shot 2024-04-02 at 2 07 06 PM
There is the following error in the terminal:
Screen Shot 2024-04-02 at 2 14 30 PM

File: ui_advanced_config.py [hunk 20]
```diff
@@ -1536,7 +1627,7 @@ def create_object(self):
                 "Object ID", "Please enter an ID for the new object."
```
Comment (sophiekummer) [inline ui_advanced_config.py:1608]: if i make a new object, the name is never added to the combobox, and it does not appear that the object is added to the json
Screen Shot 2024-04-02 at 2 16 55 PM
There is the following error in the terminal:
Screen Shot 2024-04-02 at 2 18 53 PM

**Review by sophiekummer — CHANGES_REQUESTED** [review]

File: settings/objects.json [hunk 21]
```diff
@@ -84,26 +84,40 @@
...
+        "health": 100,
+        "density": 1000,
+        "comp_ids": [
+            "0",
+            "2100",
+            "3000"
+        ],
+        "points_count": true
+    },
+    "11": {
```
Comment (sophiekummer) [inline settings/objects.json]: i think this is a testing object that isn't really meant to be included in the pr, but correct me if i'm wrong

File: settings/teams.json [hunk 22]
```diff
@@ -46,6 +46,17 @@
                 "AI_file": "AI_file4.py"
             }
         ]
+    },
+    "teammm": {
```
Comment (sophiekummer) [inline settings/teams.json]: i think this is a testing team that isn't really meant to be included in the pr, but correct me if i'm wrong

File: settings/components.json [hunk 23]
```diff
@@ -1,86 +1,85 @@
...
     },
-
     "5000": {
         "id": "5000",
         "name": "Robotic Arm",
         "ctype": "Arm",
         "max_weight": 1000,
         "max_bulk": 1000,
         "item": null
+    },
```
Comment (sophiekummer) [inline settings/components.json]: i think this is a testing component that isn't really meant to be included in the pr, but correct me if i'm wrong

File: ui_advanced_config.py [hunk 24]
```diff
@@ -528,9 +528,13 @@ def init_entry_widgets(self):
         # COMPONENT
```
Comment (sophiekummer) [inline ui_advanced_config.py:528]: there is a problem with deleting components. if i attempt, the combo box correctly deleted the id/name, but the current data remains the deleted component and the json still includes the component.
Screen Shot 2024-04-04 at 12 53 17 AM
Screen Shot 2024-04-04 at 12 53 28 AM
there is the following error in the terminal:
Screen Shot 2024-04-04 at 12 53 38 AM

File: ui_advanced_config.py [hunk 25]
```diff
@@ -545,7 +549,10 @@ def init_entry_widgets(self):
         # OBJECT
```
Comment (sophiekummer) [inline ui_advanced_config.py:549]: there is a problem with deleting objects. if i attempt, nothing happens.

File: ui_advanced_config.py (hunk 24 above)
Comment (sophiekummer) [inline ui_advanced_config.py:528]: there is a problem updating the id of a component. the component associated with the old id is still in the json and a duplicated with the new id is added in addition to the old object. the combobox does not include this duplicate.

File: ui_advanced_config.py (hunk 25 above)
Comment (sophiekummer) [inline ui_advanced_config.py:549]: there is a problem updating the id of an object. the object associated with the old id is still in the json and a duplicated with the new id is added in addition to the old object. the combobox does not include this duplicate.

**Review by sophiekummer — CHANGES_REQUESTED** [review]

File: ui_advanced_config.py (hunk 25 above)
Comment (Hodgdonc) [reply ui_advanced_config.py:549]: will do tomorrow

File: ui_advanced_config.py (hunk 24 above)
Comment (Hodgdonc) [reply ui_advanced_config.py:528]: will do tomorrow

Comment (Hodgdonc) [reply ui_advanced_config.py:528]: Fixed I think

File: ui_advanced_config.py (hunk 25 above)
Comment (Hodgdonc) [reply ui_advanced_config.py:549]: Fixed I think

Comment (Hodgdonc) [reply ui_advanced_config.py:549]: fixed?

File: ui_advanced_config.py (hunk 24 above)
Comment (Hodgdonc) [reply ui_advanced_config.py:528]: fixed?

File: ui_advanced_config.py (hunk 20 above)
Comment (Hodgdonc) [reply ui_advanced_config.py:1608]: fixed?

File: ui_advanced_config.py (hunk 19 above)
Comment (Hodgdonc) [reply ui_advanced_config.py:1555]: fixed?

File: ui_advanced_config.py (hunk 25 above)
Comment (Graham-Berry) [reply ui_advanced_config.py:549]: Seems to work for me now

File: ui_advanced_config.py (hunk 24 above)
Comment (Graham-Berry) [reply ui_advanced_config.py:528]: Changes for me now, looks good

File: ui_advanced_config.py (hunk 20 above)
Comment (Graham-Berry) [reply ui_advanced_config.py:1608]: Works for me

File: ui_advanced_config.py (hunk 19 above)
Comment (Graham-Berry) [reply ui_advanced_config.py:1555]: Works for me

File: ui_advanced_config.py (hunk 18 above)
Comment (Graham-Berry) [reply ui_advanced_config.py:1355]: I can update

File: ui_advanced_config.py (hunk 25 above)
Comment (Graham-Berry) [reply ui_advanced_config.py:549]: Old id seems to now be replaced with new ID, looks good

**Review by sophiekummer — APPROVED** [review]

**Pull request merged**